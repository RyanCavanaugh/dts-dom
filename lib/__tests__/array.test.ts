import {create, emit} from "../index"

describe("an emitted array type", () => {
    describe("with one base type", () => {
        it("matches the snapshot", () => {
            const arr = create.array('string');
            const typeAlias = create.alias('testArr', arr);

            expect(emit(typeAlias)).toMatchSnapshot();
        });
    });

    describe("with multiple types", () => {
        it("matches the snapshot", () => {
            const arr = create.array('string');
            const union = create.union(['boolean', 'string']);
            const tuple = create.array([arr, 'number', union]);
            const typeAlias = create.alias('testTuple', tuple);

            expect(emit(typeAlias)).toMatchSnapshot();
        });
    });

    describe("with a tuple as base type", () => {
        it("matches the snapshot", () => {
            const tuple = create.array(['number', 'string']);
            const arr = create.array(tuple);
            const typeAlias = create.alias('testTupleArr', arr);

            expect(emit(typeAlias)).toMatchSnapshot();
        });
    });
});
