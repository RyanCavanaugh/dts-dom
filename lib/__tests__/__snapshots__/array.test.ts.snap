// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an emitted array type with a tuple as base type matches the snapshot 1`] = `
declare type testTupleArr = ([number, string])[];


`;

exports[`an emitted array type with multiple types matches the snapshot 1`] = `
declare type testTuple = [string[], number, boolean | string];


`;

exports[`an emitted array type with one base type matches the snapshot 1`] = `
declare type testArr = string[];


`;
